drivers/bbram/bbram_it8xxx2.c

```diff
@@ -91,12 +105,16 @@ static int bbram_it8xxx2_init(const struct device *dev)
 }
 
 #define BBRAM_INIT(inst)                                                                           \
-	static const struct bbram_it8xxx2_config bbram_cfg_##inst = {                              \
-		.base_addr = DT_INST_REG_ADDR(inst),                                               \
-		.size = DT_INST_REG_SIZE(inst),                                                    \
-	};                                                                                         \
+	BBRAM_DECL_CONFIG(inst);                                                                   \
 	DEVICE_DT_INST_DEFINE(inst, bbram_it8xxx2_init, NULL, NULL, &bbram_cfg_##inst,             \
 			      PRE_KERNEL_1, CONFIG_BBRAM_INIT_PRIORITY,                            \
 			      &bbram_it8xxx2_driver_api);
 
 DT_INST_FOREACH_STATUS_OKAY(BBRAM_INIT);
+
+#ifdef CONFIG_BBRAM_IT8XXX2_EMUL
+#define BBRAM_EMUL_INIT(inst)                                                                      \
+	EMUL_DT_INST_DEFINE(inst, NULL, NULL, &bbram_cfg_##inst, NULL, &it8xxx2_emul_backend_api)
+
+DT_INST_FOREACH_STATUS_OKAY(BBRAM_EMUL_INIT);
+#endif
```

Comment:
Does this have to be here?  Can this live in the emul.c file?

Reply:
It needs a pointer to the same `bbram_cfg_##inst`

Reply:
`struct bbram_it8xxx2_config` only has `base_addr` and `size` fields.  Can the emulator just make it's own copy and fill out these values from the devicetree?

Reply:
Example of another emulator replicated devicetree info:
https://github.com/zephyrproject-rtos/zephyr/blob/b802e42c6f930d94269d41432b2baed74856f4f0/drivers/usb/bc12/emul_bc12_pi3usb9201.c#L318

Reply:
This works because `pi3usb9201_emul_data_##n` is in the same file, the issue here is that we're actually ignoring the `base_addr` from the devicetree because there's no way to actually know where there will be a free chunk of memory. Instead, when the emulator is enabled we just allocate a static buffer and use that address for the driver. The emulator needs to use the same pointer. I might be able to find this at runtime, but it seemed easier to have them allocated next to each other. The alternative would be to get the `device` pointer when declaring the emulator, then at runtime getting the `conf` pointer. I can change to that if you feel strongly about it.

Reply:
I started to leave a comment in the header file wondering if you could expose the buffer as a named object in the emulator and reference it from the device tree to get the address for the device config. It was just going to be a nit, but it would be nice if the driver and emulator could be less coupled.

Reply:
I like Abe's suggestion.  This would still require an `#ifdef CONFIG_BBRAM_IT8XXX2_EMUL` in the real driver to fetch the emulator object, but that seems like a good trade off.

Reply:
If the emulator object contained the buffer, is there a way that you could have the emulator DTS node reference the buffer and then have the device node reference that property of the emulator node? Then the "are we using the emulator" logic would be in the .dts file of the build that uses the emulator object.

Reply:
I think I got as close as I can to what you're looking for. I'm not sure how I would be able to expose the buffer as a named object in the devicetree since there might be multiple instances of it. Let me know if the latest patch works.